python/tests/expressions/test_expressions_base.py

```diff
@@ -364,3 +365,272 @@ def test_boolean_expression_visit_raise_not_implemented_error():
         base.visit("foo", visitor=visitor)
 
     assert str(exc_info.value) == "Cannot visit unsupported expression: foo"
+
+
+def test_always_true_expression_binding(table_schema_simple):
+    """Test that visiting an always-true expression returns always-true"""
+    unbound_expression = base.AlwaysTrue()
+    bound_expression = base.visit(unbound_expression, visitor=base.BindVisitor(schema=table_schema_simple))
+    assert bound_expression == base.AlwaysTrue()
+
+
+def test_always_false_expression_binding(table_schema_simple):
+    """Test that visiting an always-false expression returns always-false"""
+    unbound_expression = base.AlwaysFalse()
+    bound_expression = base.visit(unbound_expression, visitor=base.BindVisitor(schema=table_schema_simple))
+    assert bound_expression == base.AlwaysFalse()
+
+
+def test_always_false_and_always_true_expression_binding(table_schema_simple):
+    """Test that visiting both an always-true AND always-false expression returns always-false"""
+    unbound_expression = base.And(base.AlwaysTrue(), base.AlwaysFalse())
+    bound_expression = base.visit(unbound_expression, visitor=base.BindVisitor(schema=table_schema_simple))
+    assert bound_expression == base.AlwaysFalse()
+
+
+def test_always_false_or_always_true_expression_binding(table_schema_simple):
+    """Test that visiting always-true OR always-false expression returns always-true"""
+    unbound_expression = base.Or(base.AlwaysTrue(), base.AlwaysFalse())
+    bound_expression = base.visit(unbound_expression, visitor=base.BindVisitor(schema=table_schema_simple))
+    assert bound_expression == base.AlwaysTrue()
+
+
+@pytest.mark.parametrize(
+    "unbound_and_expression,expected_bound_expression",
+    [
+        (
+            base.And(
```

Comment:
Just curious, why `base.` everywhere? Personally, this introduces visual noise to me.